### ISSUE TITLE
Hunt for excess "graph translations"

### ISSUE THREAD

**Issue by jacomyal** [gephi/gephi-lite#78]
We use different data structures to represent the graph:

  - Most of the time, we have a graphology instance with no node/edge attribute for traversal, and dictionaries to handle nodes and edges attributes
  - For custom scripting, we have a graphology instance with all the nodes and edges attributes, to have a better dev experience
  - For sigma, we have a graphology instance with all data necessary to properly render the graph

We need to hunt for unnecessary translations, that might have a negative impact on performances. One first example is this one:
https://github.com/gephi/gephi-lite/blob/401feafff1f4df31ddd898272b4bf2a5257db9eb/src/core/filters/utils.ts#L90

Basically, for filtering, we always transform the dataset to the "full" graph, even when the filter is not a custom script.
